packages/shorebird_code_push_protocol/lib/src/models/full_user.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:meta/meta.dart';
+import 'package:shorebird_code_push_protocol/shorebird_code_push_protocol.dart';
+
+part 'full_user.g.dart';
+
+/// {@template user}
+/// A fully-detailed user object, possibly including sensitive information.
+/// This should only be used when querying the user's own information. For other
+/// users, use [User].
+/// {@endtemplate}
+@JsonSerializable()
+class FullUser {
```

Comment:
For consistency would it make sense to use UserMetata here similar to AppMetadata?

Reply:
Happy to adjust the naming, although what we're communicating here isn't really metadata, just a slimmed-down version of a full object.

Reply:
In that case it seems like it might be better to leave User as is and introduce a new class specifically for the new use case (I don’t think I have enough context here to suggest a name)

Reply:
The desire is to avoid accidentally leaking sensitive user info by including a User in a response object. Given a choice between PublicUser/User and User/PrivateUser, I feel like the former (i.e., the default "User" object containing no sensitive information) is less of a footgun.

Reply:
Thanks for the context! I’m fine with it just feel like the naming could be improved 🤷 

PrivateUser vs PublicUser seems better and more explicit based on the goals of this PR imo

Reply:
Done